xl2times/utils.py

```diff
@@ -315,6 +315,56 @@ def round_sig(x, sig_figs):
 default_log_name = "log" if default_log_name == "" else default_log_name
 
 
+def _is_match_on_windows(path: Path, pattern: str) -> bool:
+    """Hack for case-insensitive comparisson, because case_sensitive
+    parameter in match available from Python 3.12, but not in 3.11.
+    """
+    return PureWindowsPath(path).match(pattern)
+
+
+def is_veda_based(files: list[str]) -> bool:
+    """Determine whether the model follows Veda file structure.
+    This function does not verify file extension.
+    """
+    marker = "SysSettings.*"
+
+    matches = [file for file in files if _is_match_on_windows(Path(file), marker)]
+
+    if len(matches) == 1:
+        return True
+    elif len(matches) > 1:
+        raise ValueError(f"Only one {marker} expected. Multiple detected: {matches}")
+    else:
+        return False
+
+
+def filter_veda_filename_patterns(files: list[str]) -> list[str]:
+    """Filter files by patterns recognised by Veda."""
+    legal_paths = (
+        "BY_Trans.*",
+        "LMA*.*",
+        "Set*.*",
+        "SysSettings.*",
+        "VT_*.*",
+        "SubRES_TMPL/SubRES_*.*",
+        "SuppXLS/Demands/Dem_Alloc+Series.*",
+        "SuppXLS/Demands/ScenDem_*.*",
+        "SuppXLS/ParScenFiles/Scen_Par-*.*",
+        "SuppXLS/Scen_*.*",
+        "SuppXLS/Trades/ScenTrade_*.*",
+    )
+    # Generate a set of fiels that match the patterns
+    filtered = {
+        file
+        for file in files
+        for legal_path in legal_paths
+        if _is_match_on_windows(Path(file), legal_path)
```

Comment:
```suggestion
        if _is_match_on_windows(Path(file), legal_path)
```
if Path(file.lower()).match(legal_path.lower())

If you don't mind doing case-insensitive matching on all OSes, that'll probably work. Otherwise you'd still have to switch based on OS.